### DISCUSSION TITLE
Consideration of undrained condition in mtest

### DISCUSSION THREAD

**Discussion by yy74774** [thelfer/tfel#510] [General]
Dear Dr. Helfer,

Greetings! I have recently been working on developing an _thermo-hydro-mechanical elasto-plastic model_ using _mfront_, and I found _mfornt_ to be very convenient and clear. Currently, I am planning to further validate my model using the _mtest_ tool.

During this process, I encountered a specific issue: I was considering the undrained condition ($dm_f=0$) in my model. But for me the pore pressure $p_f$ is an external variable that needs to be given directly and not obtained by calculation. Based on my current understanding and search efforts, I have not found information on undrained condition in _mtest_.
Therefore, I would like to know if you have any suggestions or documents.

Thanks in advance for your help!
Junfeng Ren

**Comment by thelfer** [comment]
@yy74774 I am sorry, I don't have enough information to answer your question. It seems that you want to find a suitable pore pressure evolution to ensure being in the undrained condition ? Is that so ?

**Reply by yy74774** [reply]
I'm sorry for not being clear enough and I'll try to explain my question this way:
I want to know if it's possible to treat the pore pressure $p_f$ as an unknown variable (rather than an external variable) for calculation in _mtest_. 
For example, when I increase the confining pressure on a sample under isotropic condition, the pore pressure will increase due to undrained conditions $dm_f=0$. I'm wondering if I can implement this type of loading condition in _mtest_, and then I can obtain the $p_f$ by iterations so that I could compare this value with my manual calculation.

Thanks a lot for your rapid reponse !
Junfeng

**Reply by thelfer** [reply]
@yy74774 I think that you can implement this in `python` using the `mtest` module: https://thelfer.github.io/tfel/web/mtest-python.html

At each time step, you will have to set up some kind of fixed-point algorithm to determine the evolution of the pore pressure at the end of the time step.
